Add STM32L433 dtsi file

markusr opened the pull request:
Add STM32L433 dtsi file

The STM32L433 has the same layout than the STM32L432, additionally it
has GPIOD, GPIOE, I2C2, USART3 and SPI2.

Also move USART3 and SPI2 out of stm32l4.dtsi since STM32L432 does not have it.

Signed-off-by: Markus Roppelt <markus.roppelt@gmx.de>

markusr commented:
@erwango 
This is a new single pull request. I closed the old one: https://github.com/zephyrproject-rtos/zephyr/pull/11479

erwango reviewed (changes requested):
Thanks @markusr.
Besides I2C2 and GPIOS, for completeness can you add USART3 and SPI2?
Should be quite close (I expect strictly the same) to the ones available in stm32l475.dtsi.

Quick trick so you don't have to open a new PR to do these changes:
These changes could be done in the same commit, so you just have to amend your last commit (in git gui), stage the changes and commit once again.
Then you can force push your change: 
```
git push -f <zephyr push remote> <branch name>
```
In your case, could be something like:
```
git push -f https://github.com/markusr/zephyr.git l433
or
git push -f origin l433
```

markusr commented:
Yes, I had some git problems yesterday. Thank you for the hints.

USART3 and SPI2 are already defined in stm32l4.dtsi which is included from stm32l432.dtsi

erwango commented:
@markusr 

> USART3 and SPI2 are already defined in stm32l4.dtsi which is included from stm32l432.dtsi

So it appears this is actually an issue.
Could you fix it as part of you PR? Would be great thanks

erwango commented:
recheck

markusr commented:
@erwango 
I updated my commit. I moved USART3 and SPI2 out of stm32l4.dtsi since STM32L432 does not have it.

Commit:
dts: arm: st: Add STM32L433 dtsi file

The STM32L433 has the same layout than the STM32L432, additionally it
has GPIOD, GPIOE, I2C2, USART3 and SPI2.

Also move USART3 and SPI2 out of stm32l4.dtsi since STM32L432 does not
have it.

Fixes #10909.

Signed-off-by: Markus Roppelt <markus.roppelt@gmx.de>

markusr commented:
@erwango I updated my commit. Now all checks pass.

erwango reviewed (approved):
Txs!
